src/Compilers/CSharp/Test/Emit2/Semantics/ParamsCollectionTests.cs

```diff
@@ -7322,6 +7279,1521 @@ public Test(int b1, int b2, int b3)
+
+        {1} local (ref {4}, {2})
+        {{
+            {3}
+            return default;
+        }};
+    }}
+}}
+""";
+
+            VerifyParameterRefSafetyScope(template, memberName: "Test.<Test2>g__local|0_0");
+        }
+
+        [Fact]
+        public void ParameterRefSafetyScope_10_Mismatch_Overriding()
+        {
+            var src = @"
+using System;
+
+abstract class C1
+{
+    public abstract Span<long> Test1(Span<long> a);
+    public abstract Span<long> Test2(Span<long> a);
+    public abstract Span<long> Test3(Span<long> a);
+}
+
+class C2 : C1
+{
+    public override Span<long> Test1(params Span<long> a)
+        => throw null;
+
+    public override Span<long> Test2(params scoped Span<long> a)
+        => throw null;
+
+    public override Span<long> Test3(scoped Span<long> a)
+        => throw null;
+}
+";
+            var comp = CreateCompilation(src, targetFramework: TargetFramework.Net80, options: TestOptions.ReleaseDll);
+
+            Assert.Equal(comp.GetMember<MethodSymbol>("C1.Test1").Parameters.Single().EffectiveScope, comp.GetMember<MethodSymbol>("C2.Test1").Parameters.Single().EffectiveScope);
+            Assert.False(comp.GetMember<MethodSymbol>("C2.Test1").Parameters.Single().IsParams);
+            Assert.NotEqual(comp.GetMember<MethodSymbol>("C1.Test2").Parameters.Single().EffectiveScope, comp.GetMember<MethodSymbol>("C2.Test2").Parameters.Single().EffectiveScope);
+            Assert.NotEqual(comp.GetMember<MethodSymbol>("C1.Test3").Parameters.Single().EffectiveScope, comp.GetMember<MethodSymbol>("C2.Test3").Parameters.Single().EffectiveScope);
+
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void ParameterRefSafetyScope_11_Mismatch_Overriding()
+        {
+            var src = @"
+using System;
+
+abstract class C1
+{
+    public abstract Span<long> Test1(params Span<long> a);
+    public abstract Span<long> Test2(params scoped Span<long> a);
+    public abstract Span<long> Test3(scoped Span<long> a);
+}
+
+class C2 : C1
+{
+    public override Span<long> Test1(Span<long> a)
+        => throw null;
+
+    public override Span<long> Test2(Span<long> a)
+        => throw null;
+
+    public override Span<long> Test3(Span<long> a)
+        => throw null;
+}
+";
+            var comp = CreateCompilation(src, targetFramework: TargetFramework.Net80, options: TestOptions.ReleaseDll);
+
+            Assert.Equal(comp.GetMember<MethodSymbol>("C1.Test1").Parameters.Single().EffectiveScope, comp.GetMember<MethodSymbol>("C2.Test1").Parameters.Single().EffectiveScope);
+            Assert.Equal(comp.GetMember<MethodSymbol>("C1.Test2").Parameters.Single().EffectiveScope, comp.GetMember<MethodSymbol>("C2.Test2").Parameters.Single().EffectiveScope);
+            Assert.NotEqual(comp.GetMember<MethodSymbol>("C1.Test3").Parameters.Single().EffectiveScope, comp.GetMember<MethodSymbol>("C2.Test3").Parameters.Single().EffectiveScope);
+
+            comp.VerifyDiagnostics(
```

Comment:
This doesn't seem correct. Since `params` is scoped by default, we should have errors on the overrides for being unscoped. This implies that the following code will compile, which is a ref safety hole:

```cs
abstract class C1
{
    public abstract Span<long> Test1(params Span<long> a);
}
class C2 : C1
{
    public override Span<long> Test1(Span<long> a) => a;
}
```

Reply:
> This doesn't seem correct. 

That might come as a surprise, but the `params` modifier, or lack there of, is getting inherited during overriding regardless of the modifiers on the overriding method. The parameters are implicitly `params` and, therefore, they are implicitly `scope`. The asserts above actually demonstrate that the scope is identical to the overridden method for Test1 and Test2.

Reply:
Alright. I think this is something to bring up with LDM and consider enforcing `scoped` or `params` on overiddes for this scenario.